490-ce-commons/src/main/java/io/harness/ccm/msp/service/impl/ManagedAccountServiceImpl.java

```
@@ -7,19 +7,37 @@
 
 package io.harness.ccm.msp.service.impl;
 
+import io.harness.account.AccountClient;
 import io.harness.ccm.msp.dao.ManagedAccountDao;
 import io.harness.ccm.msp.dto.ManagedAccount;
 import io.harness.ccm.msp.service.intf.ManagedAccountService;
+import io.harness.exception.InvalidRequestException;
+import io.harness.ng.core.dto.AccountDTO;
+import io.harness.remote.client.CGRestUtils;
 
 import com.google.inject.Inject;
 import java.util.List;
 
 public class ManagedAccountServiceImpl implements ManagedAccountService {
   @Inject private ManagedAccountDao managedAccountDao;
+  @Inject private AccountClient accountClient;
+
+  private static final String INVALID_ACCOUNT_ID_EXCEPTION =
+      "Error is performing operation. Invalid mspAccountId/managedAccountId.";
```

We dont need "Error is performing operation". We should throw 400 status with "Invalid account id" / The account is not a managed service provider account based on which account is invalid

---

Done